include/mbgl/storage/resource.hpp

```diff
@@ -20,17 +20,42 @@ class Resource {
         SpriteJSON
     };
 
-    Resource(Kind kind_, const std::string& url_)
+    struct TileData {
+        std::string urlTemplate;
+        float pixelRatio;
```

Comment:
I know we support fractional pixel ratios when rendering, but should we always round up to the nearest integer pixel ratio when requesting assets?